### ISSUE TITLE
Increasing range of input values produces NaN weights and inability to learn

### ISSUE THREAD

**Issue by jeyendranbalakrishnan** [deeplearning4j/deeplearning4j#3346]
I just started using deeplearning4j.
I started off with a very simple problem: Learning a scalar linear function of a single scalar variable:
y = f(x) = 15 + 5 * x
For this I used a very simple linear network with no hidden layer, i.e., output layer directly connected to input with one scalar weight W and one bias term b, identity activation function, and MSE output layer loss function:
yhat = w * x + b
The parameters (w,b) were trained in the same way as the RegressionFunction.java example in the dl4j-examples package, ie., stochastic gradient, Xavier initialization, Nesterov update, 0.9 momentum, and fixed learning rate 0.01.

I chose mini-batch size = 1, 1 iteration per mini-batch, and 100 epochs.

For training data, I used the following approach:
First I fixed a scale factor s > 0.
Then the 1000 input-output training pairs were generated as:
for i = 0, 1, ...., 999:
    x_i = s * ((double) i)
    y_i = 15.0 + 5.0 * x_i
x_i was fed as the input to the network during training, with desired output y_i.
The pairs (x_i, y_i) were shuffled before being passed to the DataSetIterator, exactly as in the RegressionFunction.java example.

Following the training, the network was tested on the same training data.

I repeated this experiment for s = 0.001, 0.01, 0.02, 0.05, 1.0.

The observations were that for 0 < s <= 0.01, the network learned the correct W and b by the end of the first epoch. MSE was effectively zero.
But for s >= 0.02, the parameters and network predictions were either NaNs, or the trained network produced a constant predicted output for all training inputs { x_i }.

For scale factor s, the set of input values { x_i } is: { 0, s, ...., 999s }.
So for s = 0.01, the inputs were in the range [0, 9.99]
For s = 0.02, the inputs were in the range [0, 19.98]
etc.

I used nd4j and deeplearning4j version 0.8, on Windows 10 with quad core I7 CPU and 16GB RAM, with native platform backend. Maven central was used to pull down the jars.
In all cases, training time was less than a minute.

So my questions are: 
- Why does the training fail completely when the input values are scaled up?
- If there is a maximum and minimum range of values for the inputs, how do I find out what these are?

Thank you!
